pkg/frontend/querymiddleware/astmapper/instant_splitting_stats.go

```diff
@@ -0,0 +1,21 @@
+// SPDX-License-Identifier: AGPL-3.0-only
+
+package astmapper
+
+type InstantSplitterStats struct {
+	splitQueries int
```

Comment:
In addition to `splitQueries` we can also add a more detailed reason for a no operation. There are two possible causes:
1) range interval smaller than configured split by time
2) none of the query expressions is splittable. This counter would be more difficult to keep track off because of the recursive logic on MapNode

Reply:
Looks a great idea. We can do in a follow up PR or add it to the TODO list in the issue and keep it for later. Up to you!